libdnf/module/ModulePackageContainer.cpp

```diff
@@ -1691,4 +1704,55 @@ void ModulePackageContainer::updateFailSafeData()
     map_free(&map);
 }
 
+void ModulePackageContainer::applyEols(){
+    auto activatedModules = pImpl->activatedModules.get();
+    if (!activatedModules) {
+        return;
+    }
+    std::vector<ModulePackage *> output;
+    Id moduleId = -1;
+    while ((moduleId = activatedModules->next(moduleId)) != -1) {
+        auto modulePkg = getModulePackage(moduleId);
+        if (!isEnabled(modulePkg)) {
+            continue;
+        }
+        /* We cannot access the eol through the ModulemdModuleStream which is
+         * wrapped in modulePkg because it was created with metedata from only
+         * one repository but other repositories can have new eols which affect
+         * this module stream. We have to use merged modular metadata from all
+         * available repositories.
+         */
+        ModulemdEol *e = pImpl->moduleMetadata.getNewestEol(modulePkg);
+        if (e) {
+            if (modulemd_eol_get_reset(e)) {
+                continue;
+            }
+
+            auto moduleName = modulemd_eol_get_obsoleted_by_module_name(e);
+            auto moduleStream = modulemd_eol_get_obsoleted_by_module_stream(e);
+
+            if (moduleName && moduleStream) {
+                enable(moduleName, moduleStream, false);
```

Comment:
What about when module `moduleName` is disabled? Then I would suggest that obsolete should not be applied.

Reply:
That sounds like a "conflicting request" in transaction.
DNF should error out on that.

Reply:
But then every transaction would throw the error until the user either enables the target module, disables the obsoleted stream or the metadata are changed.